.github/workflows/check-by-name.yml

```diff
@@ -57,117 +32,47 @@ jobs:
           fetch-depth: 2
       - name: Determining PR git hashes
         run: |
-          # For pull_request_target this is the same as $GITHUB_SHA
-          echo "baseSha=$(git rev-parse HEAD^1)" >> "$GITHUB_ENV"
-
-          echo "headSha=$(git rev-parse HEAD^2)" >> "$GITHUB_ENV"
+          # Please don't put any logic here.  To ensure local
+          # reproducibility, this file should only print diagnostics
+          # (`env`, `set -x`) and execute scripts found elsewhere in
+          # nixpkgs.
+          env
+          set -x
+          exec pkgs/test/nixpkgs-check-by-name/workflows/check-by-name/determine-pr-hashes.sh
```

Comment:
This should also stay, it's GitHub-specific. Locally it's rarely the case to have a merge commit with the right parent commits.